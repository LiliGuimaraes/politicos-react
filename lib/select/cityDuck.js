'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.fetchCity = exports.changeCitySelected = exports.changeCityList = exports.CITY_SELECTED = exports.CITY_LIST = undefined;

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; /*
                                                                                                                                                                                                                                                                   * Copyright (c) 2017, Marcelo Jorge Vieira <metal@alucinados.com>
                                                                                                                                                                                                                                                                   *
                                                                                                                                                                                                                                                                   * This program is free software: you can redistribute it and/or modify
                                                                                                                                                                                                                                                                   * it under the terms of the GNU Affero General Public License as
                                                                                                                                                                                                                                                                   * published by the Free Software Foundation, either version 3 of the
                                                                                                                                                                                                                                                                   * License, or (at your option) any later version.
                                                                                                                                                                                                                                                                   *
                                                                                                                                                                                                                                                                   * This program is distributed in the hope that it will be useful,
                                                                                                                                                                                                                                                                   * but WITHOUT ANY WARRANTY; without even the implied warranty of
                                                                                                                                                                                                                                                                   * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
                                                                                                                                                                                                                                                                   * GNU Affero General Public License for more details.
                                                                                                                                                                                                                                                                   *
                                                                                                                                                                                                                                                                   * You should have received a copy of the GNU Affero General Public License
                                                                                                                                                                                                                                                                   * along with this program. If not, see <http://www.gnu.org/licenses/>.
                                                                                                                                                                                                                                                                   */

var _HTTPClient = require('../HTTPClient');

var _HTTPClient2 = _interopRequireDefault(_HTTPClient);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var CITY_LIST = exports.CITY_LIST = 'CITY_LIST';
var CITY_SELECTED = exports.CITY_SELECTED = 'CITY_SELECTED';

// Action creators
var changeCityList = exports.changeCityList = function changeCityList(list) {
  return {
    type: CITY_LIST,
    list: list
  };
};

var changeCitySelected = exports.changeCitySelected = function changeCitySelected(selected) {
  return {
    type: CITY_SELECTED,
    selected: selected
  };
};

// Side-effects / thunks

var fetchCity = exports.fetchCity = function fetchCity(selected) {
  return function (dispatch) {
    var filter = selected != '' ? '?q=' + selected : '';
    return _HTTPClient2.default.get('/cities/search/' + filter).then(function (response) {
      dispatch(changeCityList(response.data));
    });
  };
};

// Reducer
var INITIAL_STATE = {
  list: null,
  selected: null,
  query: 'candidacies__city__name__in'
};

var cityReducer = function cityReducer() {
  var state = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : INITIAL_STATE;
  var action = arguments[1];

  switch (action.type) {
    case CITY_LIST:
      return _extends({}, state, {
        list: action.list
      });
    case CITY_SELECTED:
      return _extends({}, state, {
        selected: action.selected
      });
    default:
      return state;
  }
};

exports.default = cityReducer;