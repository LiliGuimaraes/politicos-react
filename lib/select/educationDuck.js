'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

/*
 * Copyright (c) 2017, Marcelo Jorge Vieira <metal@alucinados.com>
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as
 * published by the Free Software Foundation, either version 3 of the
 * License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program. If not, see <http://www.gnu.org/licenses/>.
 */

var EDUCATION_LIST = exports.EDUCATION_LIST = 'EDUCATION_LIST';
var EDUCATION_SELECTED = exports.EDUCATION_SELECTED = 'EDUCATION_SELECTED';

// Action creators
var changeEducationList = exports.changeEducationList = function changeEducationList(list) {
  return {
    type: EDUCATION_LIST,
    list: list
  };
};

var changeEducationSelected = exports.changeEducationSelected = function changeEducationSelected(selected) {
  return {
    type: EDUCATION_SELECTED,
    selected: selected
  };
};

// Reducer
var INITIAL_STATE = {
  list: null,
  selected: null,
  query: 'education__name__in'
};

var educationReducer = function educationReducer() {
  var state = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : INITIAL_STATE;
  var action = arguments[1];

  switch (action.type) {
    case EDUCATION_LIST:
      return _extends({}, state, {
        list: action.list
      });
    case EDUCATION_SELECTED:
      return _extends({}, state, {
        selected: action.selected
      });
    default:
      return state;
  }
};

exports.default = educationReducer;