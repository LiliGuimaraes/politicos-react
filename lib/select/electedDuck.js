'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

/*
 * Copyright (c) 2017, Marcelo Jorge Vieira <metal@alucinados.com>
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as
 * published by the Free Software Foundation, either version 3 of the
 * License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program. If not, see <http://www.gnu.org/licenses/>.
 */

var ELECTED_LIST = exports.ELECTED_LIST = 'ELECTED_LIST';
var ELECTED_SELECTED = exports.ELECTED_SELECTED = 'ELECTED_SELECTED';

// Action creators
var changeElectedList = exports.changeElectedList = function changeElectedList(list) {
  return {
    type: ELECTED_LIST,
    list: list
  };
};

var changeElectedSelected = exports.changeElectedSelected = function changeElectedSelected(selected) {
  return {
    type: ELECTED_SELECTED,
    selected: selected
  };
};

// Reducer
var INITIAL_STATE = {
  list: null,
  selected: null,
  query: 'candidacies__elected__in'
};

var electedReducer = function electedReducer() {
  var state = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : INITIAL_STATE;
  var action = arguments[1];

  switch (action.type) {
    case ELECTED_LIST:
      return _extends({}, state, {
        list: action.list
      });
    case ELECTED_SELECTED:
      return _extends({}, state, {
        selected: action.selected
      });
    default:
      return state;
  }
};

exports.default = electedReducer;