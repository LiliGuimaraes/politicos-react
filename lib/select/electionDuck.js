'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

/*
 * Copyright (c) 2017, Marcelo Jorge Vieira <metal@alucinados.com>
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as
 * published by the Free Software Foundation, either version 3 of the
 * License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program. If not, see <http://www.gnu.org/licenses/>.
 */

var ELECTION_LIST = exports.ELECTION_LIST = 'ELECTION_LIST';
var ELECTION_SELECTED = exports.ELECTION_SELECTED = 'ELECTION_SELECTED';

// Action creators
var changeElectionList = exports.changeElectionList = function changeElectionList(list) {
  return {
    type: ELECTION_LIST,
    list: list
  };
};

var changeElectionSelected = exports.changeElectionSelected = function changeElectionSelected(selected) {
  return {
    type: ELECTION_SELECTED,
    selected: selected
  };
};

// Reducer
var INITIAL_STATE = {
  list: null,
  selected: null,
  query: 'candidacies__election_round__election__year__in'
};

var educationReducer = function educationReducer() {
  var state = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : INITIAL_STATE;
  var action = arguments[1];

  switch (action.type) {
    case ELECTION_LIST:
      return _extends({}, state, {
        list: action.list
      });
    case ELECTION_SELECTED:
      return _extends({}, state, {
        selected: action.selected
      });
    default:
      return state;
  }
};

exports.default = educationReducer;