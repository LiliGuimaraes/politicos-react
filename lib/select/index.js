'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.SelectState = exports.SelectPolitician = exports.SelectPoliticalParty = exports.SelectPoliticalOffice = exports.SelectOccupation = exports.SelectMaritalStatus = exports.SelectGender = exports.SelectElection = exports.SelectElected = exports.SelectEducation = exports.SelectCity = undefined;

var _SelectCity = require('./SelectCity');

var _SelectCity2 = _interopRequireDefault(_SelectCity);

var _SelectEducation = require('./SelectEducation');

var _SelectEducation2 = _interopRequireDefault(_SelectEducation);

var _SelectElected = require('./SelectElected');

var _SelectElected2 = _interopRequireDefault(_SelectElected);

var _SelectElection = require('./SelectElection');

var _SelectElection2 = _interopRequireDefault(_SelectElection);

var _SelectGender = require('./SelectGender');

var _SelectGender2 = _interopRequireDefault(_SelectGender);

var _SelectMaritalStatus = require('./SelectMaritalStatus');

var _SelectMaritalStatus2 = _interopRequireDefault(_SelectMaritalStatus);

var _SelectOccupation = require('./SelectOccupation');

var _SelectOccupation2 = _interopRequireDefault(_SelectOccupation);

var _SelectPoliticalOffice = require('./SelectPoliticalOffice');

var _SelectPoliticalOffice2 = _interopRequireDefault(_SelectPoliticalOffice);

var _SelectPoliticalParty = require('./SelectPoliticalParty');

var _SelectPoliticalParty2 = _interopRequireDefault(_SelectPoliticalParty);

var _SelectPolitician = require('./SelectPolitician');

var _SelectPolitician2 = _interopRequireDefault(_SelectPolitician);

var _SelectState = require('./SelectState');

var _SelectState2 = _interopRequireDefault(_SelectState);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

exports.SelectCity = _SelectCity2.default;
exports.SelectEducation = _SelectEducation2.default;
exports.SelectElected = _SelectElected2.default;
exports.SelectElection = _SelectElection2.default;
exports.SelectGender = _SelectGender2.default;
exports.SelectMaritalStatus = _SelectMaritalStatus2.default;
exports.SelectOccupation = _SelectOccupation2.default;
exports.SelectPoliticalOffice = _SelectPoliticalOffice2.default;
exports.SelectPoliticalParty = _SelectPoliticalParty2.default;
exports.SelectPolitician = _SelectPolitician2.default;
exports.SelectState = _SelectState2.default; /*
                                              * Copyright (c) 2017, Marcelo Jorge Vieira <metal@alucinados.com>
                                              *
                                              * This program is free software: you can redistribute it and/or modify
                                              * it under the terms of the GNU Affero General Public License as
                                              * published by the Free Software Foundation, either version 3 of the
                                              * License, or (at your option) any later version.
                                              *
                                              * This program is distributed in the hope that it will be useful,
                                              * but WITHOUT ANY WARRANTY; without even the implied warranty of
                                              * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
                                              * GNU Affero General Public License for more details.
                                              *
                                              * You should have received a copy of the GNU Affero General Public License
                                              * along with this program. If not, see <http://www.gnu.org/licenses/>.
                                              */