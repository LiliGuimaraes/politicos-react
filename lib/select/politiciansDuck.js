'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

/*
 * Copyright (c) 2017, Marcelo Jorge Vieira <metal@alucinados.com>
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as
 * published by the Free Software Foundation, either version 3 of the
 * License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program. If not, see <http://www.gnu.org/licenses/>.
 */

var POLITICIANS_CHANGE = exports.POLITICIANS_CHANGE = 'POLITICIANS_CHANGE';
var POLITICIANS_RESET = exports.POLITICIANS_RESET = 'POLITICIANS_RESET';

// Action creators
var changePoliticiansList = exports.changePoliticiansList = function changePoliticiansList(politicians) {
  return {
    type: POLITICIANS_CHANGE,
    politicians: politicians
  };
};

var resetPoliticiansList = exports.resetPoliticiansList = function resetPoliticiansList() {
  return {
    type: POLITICIANS_RESET
  };
};

// Reducer
var INITIAL_STATE = {};

var listPoliticiansReducer = function listPoliticiansReducer() {
  var state = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : INITIAL_STATE;
  var action = arguments[1];

  switch (action.type) {
    case POLITICIANS_CHANGE:
      return _extends({}, state, {
        objects: action.politicians.objects,
        meta: action.politicians.meta
      });
    case POLITICIANS_RESET:
      return _extends({}, state, {
        meta: null,
        objects: null
      });
    default:
      return state;
  }
};

exports.default = listPoliticiansReducer;