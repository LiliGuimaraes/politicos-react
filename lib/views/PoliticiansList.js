'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.PoliticiansList = undefined;

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _reactRedux = require('react-redux');

var _HTTPClient = require('../HTTPClient');

var _HTTPClient2 = _interopRequireDefault(_HTTPClient);

var _PoliticiansListItem = require('./PoliticiansListItem');

var _PoliticiansListItem2 = _interopRequireDefault(_PoliticiansListItem);

var _politiciansDuck = require('../select/politiciansDuck');

var _Loading = require('../components/Loading');

var _Loading2 = _interopRequireDefault(_Loading);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; } /*
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                * Copyright (c) 2016, Marcelo Jorge Vieira <metal@alucinados.com>
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                *
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                * This program is free software: you can redistribute it and/or modify
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                * it under the terms of the GNU Affero General Public License as
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                * published by the Free Software Foundation, either version 3 of the
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                * License, or (at your option) any later version.
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                *
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                * This program is distributed in the hope that it will be useful,
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                * but WITHOUT ANY WARRANTY; without even the implied warranty of
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                * GNU Affero General Public License for more details.
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                *
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                * You should have received a copy of the GNU Affero General Public License
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                * along with this program. If not, see <http://www.gnu.org/licenses/>.
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                */

var PoliticiansList = exports.PoliticiansList = function (_Component) {
  _inherits(PoliticiansList, _Component);

  function PoliticiansList() {
    _classCallCheck(this, PoliticiansList);

    return _possibleConstructorReturn(this, (PoliticiansList.__proto__ || Object.getPrototypeOf(PoliticiansList)).apply(this, arguments));
  }

  _createClass(PoliticiansList, [{
    key: 'componentDidMount',
    value: function componentDidMount() {
      var _this2 = this;

      _HTTPClient2.default.get('/politicians/').then(function (politicians) {
        _this2.props.dispatch((0, _politiciansDuck.changePoliticiansList)(politicians.data));
      });
    }
  }, {
    key: 'render',
    value: function render() {
      var _this3 = this;

      if (!this.props.data) {
        return _react2.default.createElement(_Loading2.default, null);
      }

      var politicianItems = this.props.data.map(function (politician) {
        return _react2.default.createElement(_PoliticiansListItem2.default, {
          onPoliticianSelect: _this3.props.onPoliticianSelect,
          key: politician.id,
          politician: politician
        });
      });

      return _react2.default.createElement(
        'ul',
        { className: 'col-lg-12 list-group politicians' },
        politicianItems
      );
    }
  }]);

  return PoliticiansList;
}(_react.Component);

/* istanbul ignore next */


var mapStateToProps = function mapStateToProps(_ref) {
  var politicians = _ref.politicians;

  return {
    data: politicians.objects
  };
};

PoliticiansList.propTypes = {
  data: _propTypes2.default.array,
  dispatch: _propTypes2.default.func,
  onPoliticianSelect: _propTypes2.default.func
};

exports.default = (0, _reactRedux.connect)(mapStateToProps)(PoliticiansList);