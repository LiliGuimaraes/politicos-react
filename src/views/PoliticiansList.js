/*
 * Copyright (c) 2016, Marcelo Jorge Vieira <metal@alucinados.com>
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as
 * published by the Free Software Foundation, either version 3 of the
 * License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program. If not, see <http://www.gnu.org/licenses/>.
 */

import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'

import { Scrollbars } from 'react-custom-scrollbars'
import { List } from 'material-ui'

import PoliticiansListItem from './PoliticiansListItem'
import { changePoliticiansList } from '../select/politiciansDuck'
import Loading from '../components/Loading'

export class PoliticiansList extends Component {
  componentDidMount() {
    this.props.HTTPClient.get('/politicians/').then(politicians => {
      this.props.dispatch(changePoliticiansList(politicians.data))
    })
  }

  render() {
    if (!this.props.data) {
      return (
        <List style={{ paddingTop: 70, marginRight: 400 }}>
          <Loading />
        </List>
      )
    }

    const politicianItems = this.props.data.map(politician => {
      return (
        <PoliticiansListItem
          onPoliticianSelect={this.props.onPoliticianSelect}
          key={politician.id}
          politician={politician}
        />
      )
    })

    return (
      <List
        style={{
          padding: 0,
          marginRight: 400,
          overflow: 'auto',
          height: '700px', // FIXME
          paddingTop: 66,
        }}
      >
        <Scrollbars autoHide autoHideTimeout={1000} autoHideDuration={200}>
          {politicianItems}
        </Scrollbars>
      </List>
    )
  }
}

/* istanbul ignore next */
const mapStateToProps = ({ politicians }) => {
  return {
    data: politicians.objects,
  }
}

PoliticiansList.propTypes = {
  data: PropTypes.array,
  dispatch: PropTypes.func,
  onPoliticianSelect: PropTypes.func,
  HTTPClient: PropTypes.object,
}

export default connect(mapStateToProps)(PoliticiansList)
