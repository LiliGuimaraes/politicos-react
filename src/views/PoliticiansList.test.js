/*
 * Copyright (c) 2017, Marcelo Jorge Vieira <metal@alucinados.com>
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as
 * published by the Free Software Foundation, either version 3 of the
 * License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program. If not, see <http://www.gnu.org/licenses/>.
 */

import React from 'react'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'

import { changePoliticiansList } from '../select/politiciansDuck'
import { PoliticiansList } from './PoliticiansList'

describe('<PoliticiansList />', () => {
  it('renders without crashing', () => {
    const HTTPClient = { get: () => Promise.resolve({}) }
    const data = [
      {
        id: 123,
        name: 'Metal',
        alternative_names: ['Marcelo'],
        political_parties: [{ political_party: { siglum: 'rock' } }],
        marital_status: { name: 'single' },
        occupation: { name: 'Developer' },
        education: { name: 'rock' },
        date_of_birth: '1800-01-01',
        nationality: { name: 'rock' },
        place_of_birth: 'heavy metal',
        state: { name: 'rock' },
        ethnicity: { name: 'rock' },
        email: 'rock@rock.com',
        website: 'https://rock.com',
      },
    ]
    const component = renderer.create(
      <PoliticiansList HTTPClient={HTTPClient} data={data} />
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders <Loading /> when there is no data', () => {
    const HTTPClient = { get: () => Promise.resolve({}) }
    const component = renderer.create(
      <PoliticiansList HTTPClient={HTTPClient} />
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  describe('.componentDidMount', () => {
    it("calls HTTPClient with 'politicians' URL", () => {
      const dispatchMock = jest.fn()
      const HTTPClient = { get: () => Promise.resolve({}) }
      const spy = jest.spyOn(HTTPClient, 'get')
      const component = shallow(
        <PoliticiansList HTTPClient={HTTPClient} dispatch={dispatchMock} />
      )

      component.instance().componentDidMount()
      expect(spy).toHaveBeenCalledWith('/politicians/')
      expect(dispatchMock).toHaveBeenCalledTimes(0)

      spy.mockReset()
      spy.mockRestore()
    })

    it('calls changePoliticiansList', () => {
      const result = { data: {} }
      const HTTPClient = {
        get: jest.fn(() => {
          return {
            then: fn => fn(result),
          }
        }),
      }
      const dispatchMock = jest.fn()
      const component = shallow(
        <PoliticiansList HTTPClient={HTTPClient} dispatch={dispatchMock} />
      )
      component.instance().componentDidMount()

      expect(dispatchMock).toHaveBeenCalledWith(
        changePoliticiansList(result.data)
      )
    })
  })
})
