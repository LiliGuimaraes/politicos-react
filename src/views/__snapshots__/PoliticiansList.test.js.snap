// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PoliticiansList /> renders <Loading /> when there is no data 1`] = `
<ul
  className="MuiList-root-1 MuiList-padding-2"
  style={
    Object {
      "marginRight": 400,
      "paddingTop": 70,
    }
  }
>
  <ul
    className="MuiList-root-1 MuiList-padding-2"
  >
    <li
      className="MuiButtonBase-root-15 MuiListItem-root-5 MuiListItem-gutters-12 MuiListItem-button-13 MuiListItem-default-8"
      disabled={false}
      onBlur={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex={0}
    >
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root-18 MuiListItemIcon-root-17"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 6v3l4-4-4-4v3c-4.42 0-8 3.58-8 8 0 1.57.46 3.03 1.24 4.26L6.7 14.8c-.45-.83-.7-1.79-.7-2.8 0-3.31 2.69-6 6-6zm6.76 1.74L17.3 9.2c.44.84.7 1.79.7 2.8 0 3.31-2.69 6-6 6v-3l-4 4 4 4v-3c4.42 0 8-3.58 8-8 0-1.57-.46-3.03-1.24-4.26z"
        />
      </svg>
      <div
        className="MuiListItemText-root-25 MuiListItemText-inset-26"
      >
        <h3
          className="MuiTypography-root-30 MuiTypography-subheading-37 MuiListItemText-text-28"
        >
          Loading...
        </h3>
      </div>
      <span
        className="MuiTouchRipple-root-54"
      />
    </li>
  </ul>
</ul>
`;

exports[`<PoliticiansList /> renders without crashing 1`] = `
<ul
  className="MuiList-root-1 MuiList-padding-2"
  style={
    Object {
      "height": "700px",
      "marginRight": 400,
      "overflow": "auto",
      "padding": 0,
      "paddingTop": 66,
    }
  }
>
  <div
    style={
      Object {
        "height": "100%",
        "overflow": "hidden",
        "position": "relative",
        "width": "100%",
      }
    }
  >
    <div
      style={
        Object {
          "WebkitOverflowScrolling": "touch",
          "bottom": 0,
          "left": 0,
          "marginBottom": 0,
          "marginRight": 0,
          "overflow": "scroll",
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    >
      <div
        style={
          Object {
            "marginLeft": 24,
            "marginTop": 48,
          }
        }
      >
        <h2>
          undefined (Metal)
        </h2>
        <li
          className="MuiListItem-root-5 MuiListItem-gutters-12 MuiListItem-default-8"
          disabled={false}
        >
          <div
            style={
              Object {
                "display": "flex",
                "width": "100%",
              }
            }
          >
            <div
              style={
                Object {
                  "flexGrow": 1,
                  "width": "33%",
                }
              }
            >
              <div>
                <span
                  aria-hidden="true"
                  className="glyphicon glyphicon-user"
                />
                Sem Foto
              </div>
            </div>
            <div
              style={
                Object {
                  "flexGrow": 1,
                  "width": "33%",
                }
              }
            >
              <p>
                Sexo: 
                Não informado
              </p>
              <p>
                Partido: 
                rock
              </p>
              <p>
                Estado civil: 
                single
              </p>
              <p>
                Ocupação: 
                Developer
              </p>
              <p>
                Escolaridade: 
                rock
              </p>
              <p>
                Data de nascimento: 
                1800-01-01
              </p>
              <p>
                Nacionalidade: 
                rock
              </p>
              <p>
                Naturalidade: 
                heavy metal/rock
              </p>
              <p>
                Nomes Alternativos: 
              </p>
              <p>
                Cor/Raça: 
                rock
              </p>
              <p>
                Email: 
                rock@rock.com
              </p>
              <p>
                Website: 
                https://rock.com
              </p>
            </div>
            <div
              style={
                Object {
                  "flexGrow": 1,
                  "width": "33%",
                }
              }
            >
              <div>
                <h3
                  className="panel-title"
                >
                  Candidaturas
                </h3>
                <div
                  className="panel-body"
                />
              </div>
            </div>
          </div>
        </li>
      </div>
    </div>
    <div
      style={
        Object {
          "borderRadius": 3,
          "bottom": 2,
          "display": "none",
          "height": 6,
          "left": 2,
          "opacity": 0,
          "position": "absolute",
          "right": 2,
          "transition": "opacity 200ms",
        }
      }
    >
      <div
        style={
          Object {
            "backgroundColor": "rgba(0,0,0,.2)",
            "borderRadius": "inherit",
            "cursor": "pointer",
            "display": "block",
            "height": "100%",
            "position": "relative",
          }
        }
      />
    </div>
    <div
      style={
        Object {
          "borderRadius": 3,
          "bottom": 2,
          "display": "none",
          "opacity": 0,
          "position": "absolute",
          "right": 2,
          "top": 2,
          "transition": "opacity 200ms",
          "width": 6,
        }
      }
    >
      <div
        style={
          Object {
            "backgroundColor": "rgba(0,0,0,.2)",
            "borderRadius": "inherit",
            "cursor": "pointer",
            "display": "block",
            "position": "relative",
            "width": "100%",
          }
        }
      />
    </div>
  </div>
</ul>
`;
